### PR TITLE
Sync checksums and fix network encoding issues.

### PR DESCRIPTION
**Pull request by num0005**
Also fixes code that didn't work with optimizations turned on and added ported fixes for loading unsupported custom maps on dedicated servers.

### CODE REVIEW

File: xlive/GSCustomMenu.cpp [hunk 1]
```diff
@@ -3763,8 +3764,19 @@ __declspec(naked) void sub_2111ab_CMLTD_nak_AccountList() {//__thiscall
 }
 
 void GSCustomMenuCall_AccountList();
-
+using namespace MapChecksumSync;
 static bool CMButtonHandler_AccountList(int button_id) {
+	// todo: better error here
+	if (MapChecksumSync::get_startup_info() != startup_state::done)
+	{
+		switch (MapChecksumSync::get_startup_info())
```
Comment (shayaantx) [inline xlive/GSCustomMenu.cpp:3772]: why are we switching on a single option?

File: xlive/H2MOD_Network.cpp [hunk 2]
```diff
@@ -529,12 +532,20 @@ int __cdecl serializeChatPacket(void* a1, int a2, int a3) {
...
 	if (network->networkCommand != NULL && network->networkCommand[0] != '\0') {
-		char* commandToWrite = (char*)(a3 + (CHAT_PACKET_SIZE - 121));
-		memset(commandToWrite, 0, 121);
-		memcpy(commandToWrite, network->networkCommand, strlen(network->networkCommand));
-		//TODO: if text is precense, don't send over the command, as this call stack is from someone sending text
-		result = getDataEncodeStringMethod()(a1, (int)"command", (int)commandToWrite, 121);
+		std::string encoded_message = base64_encode(network->networkCommand, network->command_size);
+		INT32 packet_size = encoded_message.size();
+		if (LOG_CHECK(packet_size <= CHAT_PACKET_EXTEND_SIZE))
+		{
```
Comment (shayaantx) [inline xlive/H2MOD_Network.cpp]: if you're gonna put a open brace on the else below, do same for this if, please consistent

File: xlive/H2MOD_Network.cpp [hunk 3]
```diff
@@ -598,6 +610,8 @@ void deserializePlayerAddCave() {
 	mapManager->sendMapInfoPacket();
 	//inform new players of the current advanced lobby settings
 	advLobbySettings->sendLobbySettingsPacket();
+	// send sever map checksums to client
```
Comment (shayaantx) [inline xlive/H2MOD_Network.cpp]: server*

Comment (shayaantx) [reply xlive/H2MOD_Network.cpp]: same in serializeMembershipPacket

Comment (shayaantx) [reply xlive/H2MOD_Network.cpp]: same in serializeParametersUpdatePacket

**Review by shayaantx — CHANGES_REQUESTED** [review]
see comments

File: xlive/GSCustomMenu.cpp (hunk 1 above)
Comment (num0005) [reply xlive/GSCustomMenu.cpp:3772]: I was planning to give decent error messages here but gave up.

**Review by shayaantx — APPROVED** [review]

**Pull request merged**